### PR TITLE
Smoke test for domains + slight AE rework + sel context improving

### PR DESCRIPTION
**Pull request by mfalesni**

### CODE REVIEW

**Comment by mfalesni** [comment]
`test_domain_present[Default] FAILED` is actually "good" because the Default cloned domain is disabled by default. We might change the configure script a little bit ...

**Comment by mfalesni** [comment]
@seandst @psav What would you say on the Context update?

**Comment by mfalesni** [comment]
Lint report for commit 2f4973934a1e90a62559d29b574dcb271d0e4bd4:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/automate/explorer.py:86:17: E265 block comment should start with '# '
```
*CFME QE Bot*

File: cfme/tests/automate/test_smoke.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+from cfme.automate.explorer import Domain
+
+pytestmark = [pytest.mark.smoke, pytest.mark.ignore_stream("5.2")]
+
+
+@pytest.mark.parametrize("domain_name", ["ManageIQ", "RedHat"])
+def test_domain_present(domain_name, soft_assert):
+    domain = Domain(domain_name)
+    soft_assert(domain.exists, "Domain {} does not exist!".format(domain_name))
```
Comment (dajohnso) [inline cfme/tests/automate/test_smoke.py]: We also need to verify it is not disabled by default.

**Comment by dajoRH** [comment]
Lint report for commit 97499184d2547dc1426ac9dee85dee7903239cba:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit e786b99e82d11cdcc1b9948c33ca673927d1b796:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: cfme/tests/automate/test_smoke.py (hunk 1 above)
Comment (mfalesni) [inline cfme/tests/automate/test_smoke.py]: Done.

**Comment by mfalesni** [comment]
Requesting review since I also changed stream_excluder to make possible to do like this: https://github.com/RedHatQE/cfme_tests/pull/1521/files#diff-a6afcadf909d5dcf7f94e7143327ee88R11

@psav @seandst  ?^

File: cfme/fixtures/pytest_selenium.py [hunk 2]
```diff
@@ -740,6 +740,28 @@ def go_to_fixture(fixtureconf, browser):
     force_navigate(page_name)
 
 
+class ContextWrapper(object):
```
Comment (seandst) [inline cfme/fixtures/pytest_selenium.py]: This class seems evil.

If all __getattr__ is going to do is raise an exception, it really should raise AttributeError. But it's looking at something's Keys, and so it should probably raise KeyError, but the interface is an item lookup, so nobody should be doing attribute accesses in the first place.

I see the desire to list the available keys in the exception message, but it seems like subclassing dict, or wrapping a dict and implementing `__iter__` and `__contains__` would be more pythonic, especially if the correct exceptions are raised on `__getitem__` misses and the `__getattr__` override goes away.

Setting self.__dict__ to a passed in also seems like unnecessary roughness, and again points to subclassing dict (`self.update(data)`) or wrapping a dict (`self._data = data`) as more appropriate solutions.

Comment (mfalesni) [inline cfme/fixtures/pytest_selenium.py]: OK, I will try to find a better solution :)

**Comment by dajoRH** [comment]
Lint report for commit 141e876dfb1616bef81f3b0b118ef4946a662f85:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by mfalesni** [comment]
@seandst I changed the class to inherit from dict so it is now dict with .attr access enabled + it dumps all the keys when the key was not found in the context.

+ I went through cfme_tests and removed occurences of wrong context usage (User object as context or so)

**Pull request merged**